commercial/app/model/commercial/events/Event.scala

```diff
@@ -0,0 +1,206 @@
+            description: String,
+            status: String,
+            venue: Venue,
+            tickets: Seq[Option[Ticket]],
+            capacity: Int): Event = {
+
+    new Event(
+      id,
+      name,
+      startDate,
+      url,
+      description,
+      None,
+      status,
+      venue,
+      tickets.flatten,
+      capacity,
+      firstParagraph = extractFirstParagraph(description)
+    )
+  }
+
+  private lazy val dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ"
+  implicit val jodaDateTimeReads: Reads[DateTime] = Reads.jodaDateReads(dateFormat)
+  implicit val eventReads: Reads[Event] = (
+
+    (JsPath \ "id").read[String] and
+    (JsPath \ "name" \ "text").read[String] and
+    (JsPath \ "start" \ "utc").read[DateTime] and
+    (JsPath \ "url").read[String] and
+    (JsPath \ "description" \ "html").read[String] and
+    (JsPath \ "status").read[String] and
+    (JsPath \ "venue").read[Venue] and
+    (JsPath \ "ticket_classes").read[Seq[Option[Ticket]]] and
+    (JsPath \ "capacity").read[Int]
+    )(Event.apply(_: String, _: String, _: DateTime, _:String, _: String, _: String, _: Venue, _: Seq[Option[Ticket]], _:Int))
+}
+
+case class Masterclass(id: String,
+                       name: String,
+                       startDate: DateTime,
+                       url: String,
+                       description: String,
+                       status: String,
+                       venue: Venue,
+                       tickets: Seq[Ticket],
+                       capacity: Int,
+                       guardianUrl: String,
+                       firstParagraph: String,
+                       keywordIdSuffixes: Seq[String],
+                       mainPicture: Option[ImageElement]) extends TicketHandler {
+
+  lazy val readableDate = DateTimeFormat.forPattern("d MMMMM yyyy").print(startDate)
+
+  lazy val truncatedFirstParagraph = StringUtils.abbreviate(firstParagraph, 250)
+}
+
+object Masterclass {
+  private val guardianUrlLinkText = "Full course and returns information on the Masterclasses website"
+
+  def apply(event: Event): Masterclass = {
+
+    def extractGuardianUrl: String = {
+      val doc = Jsoup.parse(event.description)
+      val elements :Array[Element] = doc.select(s"a[href^=http://www.theguardian.com/]:contains($guardianUrlLinkText)")
+          .toArray(Array.empty[Element])
+
+      elements.headOption match {
+        case Some(e) => e.attr("href")
+        case _ => ""
+      }
+    }
+
+    new Masterclass(
+      id = event.id,
+      name = event.name,
+      startDate = event.startDate,
+      url = event.url,
+      description = event.description,
+      status = event.status,
+      venue = event.venue,
+      tickets = event.tickets,
+      capacity = event.capacity,
+      guardianUrl = extractGuardianUrl,
+      firstParagraph = event.firstParagraph,
+      keywordIdSuffixes = Nil,
+      mainPicture = None
+    )
+  }
+}
+
+trait TicketHandler{
+  def tickets: Seq[Ticket]
+  def status: String
```

Comment:
`status` probably doesn't belong here.  I think it's used to filter events out of the list.